### PR TITLE
fix(error-overlay): strip line+column from webpack internal frames

### PR DESCRIPTION
**Pull request by balazsorban44**
### What?

Follow-up on #62101

Before:
image

After:
image

### Why?

Line and column numbers are incorrect in these cases, so it is just noise to the user

### How?

if the frame file indicates a webpack internal file, do not append the line/column numbers during formatting.

Closes NEXT-2516
[Slack thread](https://vercel.slack.com/archives/C04DUD7EB1B/p1708092503998689)

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| buildDuration | 13.6s | 13.7s | N/A |
| buildDurationCached | 8.4s | 6.3s | N/A |
| nodeModulesSize | 198 MB | 198 MB | ⚠️ +2.5 kB |
| nextStartRea..uration (ms) | 435ms | 425ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 30.5 kB | 30.5 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | N/A |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 243 B | 243 B | ✓ |
| main-HASH.js gzip | 32.2 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 45.6 kB | 45.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.21 kB | 4.21 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.57 kB | 6.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| _buildManifest.js gzip | 481 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| index.html gzip | 528 B | 529 B | N/A |
| link.html gzip | 539 B | 541 B | N/A |
| withRouter.html gzip | 523 B | 523 B | ✓ |
| Overall change | 523 B | 523 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.1 kB | 95.1 kB | N/A |
| page.js gzip | 3.06 kB | 3.07 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 627 B | 624 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 990 B | 990 B | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 171 kB | 171 kB | ✓ |
| app-page-exp..prod.js gzip | 96.5 kB | 96.5 kB | ✓ |
| app-page-tur..prod.js gzip | 98.3 kB | 98.3 kB | ✓ |
| app-page-tur..prod.js gzip | 92.7 kB | 92.7 kB | ✓ |
| app-page.run...dev.js gzip | 149 kB | 149 kB | ✓ |
| app-page.run..prod.js gzip | 91.2 kB | 91.2 kB | ✓ |
| app-route-ex...dev.js gzip | 21.3 kB | 21.3 kB | ✓ |
| app-route-ex..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| app-route.ru...dev.js gzip | 21 kB | 21 kB | ✓ |
| app-route.ru..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.52 kB | 9.52 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.8 kB | 9.8 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.52 kB | 9.52 kB | ✓ |
| pages-turbo...prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| pages.runtim...dev.js gzip | 22.9 kB | 22.9 kB | ✓ |
| pages.runtim..prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| server.runti..prod.js gzip | 50.5 kB | 50.5 kB | ✓ |
| Overall change | 948 kB | 948 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/strip-line-column-webpack-internals | Change |
| - | - | - | - |
| 0.pack gzip | 1.56 MB | 1.56 MB | N/A |
| index.pack gzip | 106 kB | 105 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 6c3aa604e6594aa5d21df3ccf23e6261b3cd7087

**Comment by ijjk** [comment]
## Failing test suites 
Commit: c779bfc5f4bcb97fedb81efdbb05c11ae1fe5281

`pnpm test-dev test/e2e/middleware-matcher/index.test.ts`
- using a single matcher with i18n > adds the header for a matched path
- using a single matcher with i18n > adds the header for a mathed root path with /index
- using a single matcher with i18n > does not add the header for an unmatched path

<details>
<summary>Expand output</summary>

  ● using a single matcher with i18n › adds the header for a matched path

    expect(received).toContain(expected) // indexOf

    Expected substring: "(en) Hello from /"
    Received string:    "<!DOCTYPE html><html lang=\"en\"><head><style data-next-hide-fouc=\"true\">body{display:none}</style><noscript data-next-hide-fouc=\"true\"><style>body{display:block}</style></noscript><meta charSet=\"utf-8\"/><meta name=\"viewport\" content=\"width=device-width\"/><meta name=\"next-head-count\" content=\"2\"/><noscript data-n-css=\"\"></noscript><script defer=\"\" nomodule=\"\" src=\"/_next/static/chunks/polyfills.js\"></script><script src=\"/_next/static/chunks/fallback/webpack.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/main.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_app.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_error.js\" defer=\"\"></script><noscript id=\"__next_css__DO_NOT_USE__\"></noscript></head><body><div id=\"__next\"></div><script src=\"/_next/static/chunks/fallback/react-refresh.js\"></script><script id=\"__NEXT_DATA__\" type=\"application/json\">{\"props\":{\"pageProps\":{\"statusCode\":500}},\"page\":\"/_error\",\"query\":{},\"buildId\":\"development\",\"isFallback\":false,\"err\":{\"name\":\"TypeError\",\"source\":\"server\",\"message\":\"Cannot read properties of null (reading 'useReducer')\",\"stack\":\"TypeError: Cannot read properties of null (reading 'useReducer')\\n    at useReducer (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/react@18.2.0/node_modules/react/cjs/react.development.js:1626:21)\\n    at useErrorOverlayReducer (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/client/components/react-dev-overlay/shared.js:87:34)\\n    at ReactDevOverlay (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/client/components/react-dev-overlay/pages/ReactDevOverlay.js:31:66)\\n    at ReactDevOverlay (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/server/dev/next-dev-server.js:100:12)\\n    at renderWithHooks (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)\\n    at renderIndeterminateComponent (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)\\n    at renderElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)\\n    at renderNodeDestructiveImpl (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)\\n    at renderNodeDestructive (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)\\n    at renderNode (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)\\n    at renderHostElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3)\\n    at renderElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5952:5)\\n    at renderNodeDestructiveImpl (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)\\n    at renderNodeDestructive (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)\\n    at renderIndeterminateComponent (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5785:7)\"},\"gip\":true,\"locale\":\"en\",\"locales\":[\"en\",\"es\"],\"defaultLocale\":\"en\",\"scriptLoader\":[]}</script></body></html>"

      378 |   it(`adds the header for a matched path`, async () => {
      379 |     const res1 = await fetchViaHTTP(next.url, `/`)
    > 380 |     expect(await res1.text()).toContain(`(en) Hello from /`)
          |                               ^
      381 |     expect(res1.headers.get('X-From-Middleware')).toBe('true')
      382 |     const res2 = await fetchViaHTTP(next.url, `/es`)
      383 |     expect(await res2.text()).toContain(`(es) Hello from /`)

      at Object.toContain (e2e/middleware-matcher/index.test.ts:380:31)

  ● using a single matcher with i18n › adds the header for a mathed root path with /index

    expect(received).toContain(expected) // indexOf

    Expected substring: "(en) Hello from /"
    Received string:    "<!DOCTYPE html><html lang=\"en\"><head><style data-next-hide-fouc=\"true\">body{display:none}</style><noscript data-next-hide-fouc=\"true\"><style>body{display:block}</style></noscript><meta charSet=\"utf-8\"/><meta name=\"viewport\" content=\"width=device-width\"/><meta name=\"next-head-count\" content=\"2\"/><noscript data-n-css=\"\"></noscript><script defer=\"\" nomodule=\"\" src=\"/_next/static/chunks/polyfills.js\"></script><script src=\"/_next/static/chunks/fallback/webpack.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/main.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_app.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_error.js\" defer=\"\"></script><noscript id=\"__next_css__DO_NOT_USE__\"></noscript></head><body><div id=\"__next\"></div><script src=\"/_next/static/chunks/fallback/react-refresh.js\"></script><script id=\"__NEXT_DATA__\" type=\"application/json\">{\"props\":{\"pageProps\":{\"statusCode\":500}},\"page\":\"/_error\",\"query\":{},\"buildId\":\"development\",\"isFallback\":false,\"err\":{\"name\":\"TypeError\",\"source\":\"server\",\"message\":\"Cannot read properties of null (reading 'useReducer')\",\"stack\":\"TypeError: Cannot read properties of null (reading 'useReducer')\\n    at useReducer (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/react@18.2.0/node_modules/react/cjs/react.development.js:1626:21)\\n    at useErrorOverlayReducer (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/client/components/react-dev-overlay/shared.js:87:34)\\n    at ReactDevOverlay (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/client/components/react-dev-overlay/pages/ReactDevOverlay.js:31:66)\\n    at ReactDevOverlay (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/server/dev/next-dev-server.js:100:12)\\n    at renderWithHooks (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)\\n    at renderIndeterminateComponent (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)\\n    at renderElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)\\n    at renderNodeDestructiveImpl (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)\\n    at renderNodeDestructive (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)\\n    at renderNode (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)\\n    at renderHostElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3)\\n    at renderElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5952:5)\\n    at renderNodeDestructiveImpl (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)\\n    at renderNodeDestructive (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)\\n    at renderIndeterminateComponent (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5785:7)\"},\"gip\":true,\"locale\":\"en\",\"locales\":[\"en\",\"es\"],\"defaultLocale\":\"en\",\"scriptLoader\":[]}</script></body></html>"

      387 |   it('adds the header for a mathed root path with /index', async () => {
      388 |     const res1 = await fetchViaHTTP(next.url, `/index`)
    > 389 |     expect(await res1.text()).toContain(`(en) Hello from /`)
          |                               ^
      390 |     expect(res1.headers.get('X-From-Middleware')).toBe('true')
      391 |     const res2 = await fetchViaHTTP(next.url, `/es/index`)
      392 |     expect(await res2.text()).toContain(`(es) Hello from /`)

      at Object.toContain (e2e/middleware-matcher/index.test.ts:389:31)

  ● using a single matcher with i18n › does not add the header for an unmatched path

    expect(received).toContain(expected) // indexOf

    Expected substring: "Hello from /about/me"
    Received string:    "<!DOCTYPE html><html lang=\"en\"><head><style data-next-hide-fouc=\"true\">body{display:none}</style><noscript data-next-hide-fouc=\"true\"><style>body{display:block}</style></noscript><meta charSet=\"utf-8\"/><meta name=\"viewport\" content=\"width=device-width\"/><meta name=\"next-head-count\" content=\"2\"/><noscript data-n-css=\"\"></noscript><script defer=\"\" nomodule=\"\" src=\"/_next/static/chunks/polyfills.js\"></script><script src=\"/_next/static/chunks/fallback/webpack.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/main.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_app.js\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_error.js\" defer=\"\"></script><noscript id=\"__next_css__DO_NOT_USE__\"></noscript></head><body><div id=\"__next\"></div><script src=\"/_next/static/chunks/fallback/react-refresh.js\"></script><script id=\"__NEXT_DATA__\" type=\"application/json\">{\"props\":{\"pageProps\":{\"statusCode\":500}},\"page\":\"/_error\",\"query\":{},\"buildId\":\"development\",\"isFallback\":false,\"err\":{\"name\":\"TypeError\",\"source\":\"server\",\"message\":\"Cannot read properties of null (reading 'useReducer')\",\"stack\":\"TypeError: Cannot read properties of null (reading 'useReducer')\\n    at useReducer (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/react@18.2.0/node_modules/react/cjs/react.development.js:1626:21)\\n    at useErrorOverlayReducer (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/client/components/react-dev-overlay/shared.js:87:34)\\n    at ReactDevOverlay (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/client/components/react-dev-overlay/pages/ReactDevOverlay.js:31:66)\\n    at ReactDevOverlay (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/.pnpm/file+..+next-repo-cbc846e9ce1624a5bbdd8ec19638c12e727ab357cfc9e43db2e7d9dcd2d66f98+packages+n_gv6opbiu2yt2r3yzhpma5nf2ye/node_modules/next/dist/server/dev/next-dev-server.js:100:12)\\n    at renderWithHooks (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)\\n    at renderIndeterminateComponent (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)\\n    at renderElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)\\n    at renderNodeDestructiveImpl (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)\\n    at renderNodeDestructive (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)\\n    at renderNode (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)\\n    at renderHostElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3)\\n    at renderElement (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5952:5)\\n    at renderNodeDestructiveImpl (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)\\n    at renderNodeDestructive (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)\\n    at renderIndeterminateComponent (/tmp/next-install-255b5bab91e76c867589b0a0ed2bf9fbd87b6cb750668c9e1abd0dc62b05eca6/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5785:7)\"},\"gip\":true,\"locale\":\"en\",\"locales\":[\"en\",\"es\"],\"defaultLocale\":\"en\",\"scriptLoader\":[]}</script></body></html>"

      418 |   it(`does not add the header for an unmatched path`, async () => {
      419 |     const response = await fetchViaHTTP(next.url, `/about/me`)
    > 420 |     expect(await response.text()).toContain('Hello from /about/me')
          |                                   ^
      421 |     expect(response.headers.get('X-From-Middleware')).toBeNull()
      422 |   })
      423 | })

      at Object.toContain (e2e/middleware-matcher/index.test.ts:420:35)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/client/components/react-dev-overlay/internal/helpers/stack-frame.ts [hunk 1]
```diff
@@ -102,30 +102,35 @@ export function getOriginalStackFrames(
   )
 }
 
+const webpackRegExes = [
+  /^webpack-internal:\/\/\/(\.)?(\((\w+)\))?/,
+  /^(webpack:\/\/\/(\.)?|webpack:\/\/(_N_E\/)?)(\((\w+)\))?/,
+]
+
+function isInternal(file: string) {
```
Comment (huozhi) [inline packages/next/src/client/components/react-dev-overlay/internal/helpers/stack-frame.ts:110]:
```suggestion
function isWebpackBundled(file: string) {
```

File: packages/next/src/client/components/react-dev-overlay/internal/helpers/stack-frame.ts [hunk 2]
```diff
@@ -102,30 +102,35 @@ export function getOriginalStackFrames(
...
  * webpack://_N_E/./src/hello.tsx => ./src/hello.tsx
  * webpack://./src/hello.tsx => ./src/hello.tsx
  * webpack:///./src/hello.tsx => ./src/hello.tsx
- *
  */
 function formatFrameSourceFile(file: string) {
-  return file
-    .replace(/^webpack-internal:\/\/\/(\.)?(\((\w+)\))?/, '')
-    .replace(/^(webpack:\/\/\/(\.)?|webpack:\/\/(_N_E\/)?)(\((\w+)\))?/, '')
+  return file.replace(webpackRegExes[0], '').replace(webpackRegExes[1], '')
```
Comment (huozhi) [inline packages/next/src/client/components/react-dev-overlay/internal/helpers/stack-frame.ts:122]: If we add new regex or adjust the length later we dont need to change this part
```suggestion
  for (const regex of webpackRegExes) file = file.replace(regex, '')
  return file
```

**Review by huozhi — APPROVED** [review]

**Pull request merged**